### PR TITLE
Make paasta local-run lock on pulls

### PR DESCRIPTION
**Pull request by solarkennedy**
Although I hate to do it, it looks like we can't trust docker to pull 2 images simultaneously :(

### CODE REVIEW

File: paasta_tools/cli/cmds/local_run.py [hunk 1]
```diff
@@ -418,15 +419,17 @@ def docker_pull_image(docker_url):
...
     DEVNULL = open(os.devnull, 'wb')
-    ret, output = _run('docker pull %s' % docker_url, stream=True, stdin=DEVNULL)
-    if ret != 0:
-        paasta_print(
-            "\nPull failed. Are you authorized to run docker commands?",
-            file=sys.stderr,
-        )
-        sys.exit(ret)
+    with open('/tmp/paasta-local-run-pull.lock', 'w') as f:
+        with timed_flock(f, seconds=1800):
```
Comment (matthewbentley) [inline paasta_tools/cli/cmds/local_run.py]: Would it be useful to do something other than just throw a `TimeoutError` exception here?

File: paasta_tools/cli/cmds/local_run.py [hunk 2]
```diff
@@ -418,15 +419,17 @@ def docker_pull_image(docker_url):
     bindings due to the docker auth/registry transition. Once we are past Docker 1.6
     we can use better credential management, but for now this function assumes the
     user running the command has already been authorized for the registry"""
-    paasta_print("Please wait while the image (%s) is pulled..." % docker_url, file=sys.stderr)
+    paasta_print("Please wait while the image (%s) is pulled (times out after 30m)..." % docker_url, file=sys.stderr)
```
Comment (nhandler) [inline paasta_tools/cli/cmds/local_run.py]: Can you move the timeout to a variable that you reference here and in the call to `timed_flock`? That way, they stay in sync even when we update the timeout in the future.

File: paasta_tools/cli/cmds/local_run.py (hunk 1 above)
Comment (nhandler) [reply paasta_tools/cli/cmds/local_run.py]: A retry might make sense. Especially in the case of a Jenkins local-run for an itest, that would prevent the healthcheck from failing intermittently.

Comment (solarkennedy) [reply paasta_tools/cli/cmds/local_run.py]: Why? If we are going to retry, why not just wait longer?

If we would retry 10times, then I'll just make the timeout be 18000 instead

**Comment by solarkennedy** [comment]
I agree ideally I would fix docker, but in practice I'm not. 

You say this might break non-interactive scripts, but why doesn't them being non-interactive make a difference? The thing is they (the non-interactive use-cases) are breaking *now* because of this issue, the PR is to fix them!

File: paasta_tools/cli/cmds/local_run.py (hunk 1 above)
Comment (nhandler) [reply paasta_tools/cli/cmds/local_run.py]: You are right

**Pull request merged**